File: HKDConvertorJava_MVI/app/src/main/java/com/example/hkdconvertorjava/intent/MainIntent.java
package com.example.hkdconvertorjava.intent;

public class MainIntent {

    // Intent representing a user action to convert currency
    public static class ConvertCurrency {
        public final String hkdAmount;  // The HKD amount to be converted

        // Constructor to initialize the HKD amount
        public ConvertCurrency(String hkdAmount) {
            this.hkdAmount = hkdAmount;
        }
    }
}

File: HKDConvertorKotlin_MVI/app/src/main/java/com/example/hkdconvertorkotlin/MainIntent.kt
package com.example.hkdconvertorkotlin

// This sealed class represents the user's intent (actions).
// It defines what kind of actions the user can trigger.
sealed class MainIntent {
    // This intent represents the action to convert a currency.
    // It carries the HKD amount input by the user.
    data class ConvertCurrency(val hkdAmount: String) : MainIntent()
}